src/Repository/DocumentRepository.php

```diff
@@ -57,11 +51,9 @@ class DocumentRepository implements ObjectRepository, Selectable
      * @param ClassMetadata   $classMetadata The class metadata.
      * @phpstan-param ClassMetadata<T>   $classMetadata The class metadata.
      */
-    public function __construct(DocumentManager $dm, UnitOfWork $uow, ClassMetadata $classMetadata)
+    public function __construct(protected DocumentManager $dm, protected UnitOfWork $uow, ClassMetadata $classMetadata)
```

Comment:
You can use CPP for `$classMetadata`.
```suggestion
    public function __construct(protected DocumentManager $dm, protected UnitOfWork $uow, protected ClassMetadata $classMetadata)
```

Reply:
Unfortunately, the constructor argument name does not match the property name. To reduce the BC impact, I decided to keep it this way.